.github/workflows/run-tests.yml

```diff
@@ -174,6 +174,55 @@ jobs:
         run: |
           ./mvnw -B "-Dbuild.repository.owner=liquibase" "-Dbuild.repository.name=liquibase" "-Dbuild.branch=${{ needs.setup.outputs.thisBranchName }}" "-Dbuild.number=${{ github.run_number }}" "-Dbuild.commit=${{ needs.setup.outputs.latestMergeSha }}" "-DtrimStackTrace=false" -P 'skip-integration-tests' clean test package surefire-report:report
 
+      - name: Check Unit Test Count for Java 8, 11, 17, 21 Ubuntu
+        id: test-count-java-ubuntu
+        if: matrix.os != 'macos-13' && matrix.os != 'windows-2019'
+        run: |
+          if [[ "${{ matrix.os }}" != "macos-13" ]]; then
+            TEST_COUNT=$(find . -name '*.txt' | xargs grep -oP 'Tests run: \K[0-9]+' | awk -F ':' '{s+=$2} END {print(s)}')
+          fi
+          echo "Total Unit tests run: $TEST_COUNT"
+          if [ "$TEST_COUNT" -eq 0 ]; then
+            echo "No Unit tests were run."
+            exit 1
+          fi
+          echo "TEST_COUNT_JAVA_UBUNTU=$TEST_COUNT" >> $GITHUB_ENV
+  
+      - name: Check Unit Test Count for Java 8 macos-13
+        id: test-count-java-macos
+        if: matrix.os == 'macos-13'
+        shell: bash
+        run: |
+          TEST_COUNT=$(grep -r -o -E 'Tests run: [0-9]+' . | awk -F: '{s+=$NF} END {print s}')
+          echo "Total Unit tests run: $TEST_COUNT"
+          if [ -z "$TEST_COUNT" ] || [ "$TEST_COUNT" -eq 0 ]; then
+            echo "No test result files found or test count is zero."
+            exit 1
+          fi  
+          echo "TEST_COUNT_JAVA_MACOS=$TEST_COUNT" >> $GITHUB_ENV
+
+      - name: Check Test Count for windows-2019
+        id: test-count-java-windows
+        if: matrix.os == 'windows-2019'
+        shell: pwsh
+        run: |
+          $TEST_COUNT = (Get-ChildItem -Recurse -Filter "*.txt" | Select-String -Pattern 'Tests run: (\d+)' | ForEach-Object { $_.Matches.Groups[1].Value } | Measure-Object -Sum).Sum
+          Write-Output "Total Unit tests run: $TEST_COUNT"
+          if ($TEST_COUNT -eq 0) {
+            Write-Output "No Unit tests were run."
+            exit 1
+          }
+          echo "TEST_COUNT_JAVA_WINDOWS=$TEST_COUNT" | Out-File -Append -Encoding utf8 -FilePath $env:GITHUB_ENV
+
+      - name: Notify slack if Test Count decreases
+        if: env.TEST_COUNT_JAVA_WINDOWS && env.TEST_COUNT_JAVA_WINDOWS < 5999 || env.TEST_COUNT_JAVA_UBUNTU && env.TEST_COUNT_JAVA_UBUNTU < 5981 || env.TEST_COUNT_JAVA_MACOS && env.TEST_COUNT_JAVA_MACOS < 5981
```

Comment:
Having those hardcoded counts here may be a problem, as we will need to update them from time to time. 
Ideally we would use an action similar to https://github.com/marketplace/actions/set-persistent-value , but who knows where it stores information. 
As an option, maybe we could store this info in a text file and update it when it runs on master? 
Any other ideas @StevenMassaro ?

Reply:
the action https://github.com/marketplace/actions/set-persistent-value is not well maintained. I'd prefer using json instead.

Reply:
we could store the values in github secrets